### DISCUSSION TITLE
Crowdfunding to solve issues

### DISCUSSION THREAD

**Discussion by Whip** [community/community#17526] [Other Feature Feedback, Questions, & Ideas]
You know how some kind souls creates software for people to use for free but then don't have the time/motivation to support it. What if there was a way where everyone could donate towards a specific issue giving author an incentive to fix the issue. The more people it affects, more people can donate towards it.

Maybe the author can set a goal like $100. When we reach the goal, the author must fix that issue.

Playing devil's advocate: I hope people don't misuse it and create intentional bugs to get paid for fixing them.

**Comment by jlord** [comment]
Thanks for this feedback! Wanted to check on your expectations for how a feature like this might work. A few questions for you ✨ 

- What is your expectation if a bug takes longer to fix than expected and the $100 goal is not sufficient?
- What do you expect to happen if a solution doesn't do everything the crowdfunders hoped it would?
- What do you expect to happen when something doesn't get solved?

**Reply by Whip** [reply]
Good questions. I hope other people jump into it with their opinions as well. 

- This maybe a result of poor planning I suppose. A competent developer should be able to assess the situation properly and choose appropriate goals for time and compensation. That being said, if it does still happen and lets say it happens with me, I would follow through on my commitments and fix the issue at the goal amount _I_ picked. As for the time, I would apologize profusely 😛 More on this in the 3rd point.
- Github issues are supposed to be specific, reproducible and testable things that don't work as intended. And when money is involved, the issue should be documented very well so everyone's on the same page - as to what is broken, what is happening and what is supposed to happen - before we start.
- Perhaps, Github should keep the money in an Escrow account until the issue is resolved. Here's how it might go: author fixes the problem and closes the issue. Github gives it a 1 week waiting period during which contributors can challenge that the issue isn't actually solved. Github intervenes and tries to resolve the situation. If Github can't, the money collected is refunded to the people. Similarly, if the issue is not resolved at all and delayed by a lot, contributors can request Github to step in. When money is involved, the authors have to be held accountable. If everything goes well, 1 week later the goal amount is released to the author.

**Reply by Ayanmullick** [reply]
1. GitHub could introduce a rating system for sponsors and developers to help someone assess the probability of a dispute. 
2. GitHub should create issue template guidelines for repositories enabling crowdfunding of issues.
3. It would be convenient to be on one platform rather than using other platforms for crowdfunding issues. E.g., https://gitpay.me/#/task/607
4. GitHub should publish guidelines on structuring acceptance criteria to avoid conflict.
5. The fund could be transferred once the sponsor closes the issue.
6. Like other freelancing platforms, the funds could be sent back to the sponsor if an issue isn't resolved or if a developer rejects the issue.

**Comment by PFython** [comment]
@whip @jlord @Ayanmullick @jonespm

Great minds think alike :) I just found this after submitting a similar suggestion today, with a slightly different proposed solution.  Please have a look, share, and comment positively if you'd still like to see something like this in Github still?

https://github.com/orgs/community/discussions/61265

All the best,
Pete
